File: src/main/java/pairtrix/domain/TeamMembers.java
package pairtrix.domain;

import java.util.*;

public class TeamMembers {

    List<String> teamMembers = new ArrayList<>();

    public List<String> withSeed(Random random) {
        List<String> randomizedList = new ArrayList<>(teamMembers);
        Collections.shuffle(randomizedList, random);
        return randomizedList;
    }

    public void add(String name) {
        teamMembers.add(name);
    }
}

File: src/main/java/pairtrix/domain/TeamMembers.kt
package pairtrix.domain

import java.util.*

class TeamMembers(private val teamRepository: TeamMembersRepository) {

    fun withSeed(random: Random): List<String> {
        val randomizedList = ArrayList(teamRepository.findAll())
        Collections.shuffle(randomizedList, random)
        return randomizedList
    }

    fun add(name: String) {
        teamRepository.addMember(name)
    }
}
